tests/pytests/test_cert_framework_basic.py

```diff
@@ -76,6 +76,45 @@ def test_cf_cc_trust_data():
 
     libcf.delete_cc_trust_data(cctd)
 
+# ##############################################################################
+def test_cf_cc_trust_data_add_or_update_new_domain():
+    """
+    Exercise add_or_update_new_domain(), which will create a new certified_domain()
+    object. Dismantiling this cc_trust_data() will test the destructor of that
+    object which should correctly release memory for new certified-domains.
+    """
+    cctd = libcf.new_cc_trust_data()
+
+    result = libcf.cc_trust_data_add_or_update_new_domain(cctd,
+                                                          'test-security-domain',
+                                                          'test-dummy-certificate',
+                                                          'localhost', 8121,
+                                                          'localhost', 8123)
+    assert result is True
+
+    libcf.delete_cc_trust_data(cctd)
+
+# ##############################################################################
+def test_cf_cc_trust_data_certify_secondary_domain_not_found():
+    """
+    Exercise certify_secondary_domain(). (Verifies fix to handle a secondary
+    domain that is not found; leading to null domain ptr.)
+    """
+    cctd = libcf.new_cc_trust_data()
+
+    result = libcf.cc_trust_data_add_or_update_new_domain(cctd,
+                                                          'test-security-domain',
+                                                          'test-dummy-certificate',
+                                                          'localhost', 8121,
+                                                          'localhost', 8123)
+    assert result is True
+
+    result = libcf.cc_trust_data_certify_secondary_domain(cctd,
```

Comment:
This lines triggers, and verifies the fix for, the seg-fault in this `certify_secondary_domain()` method.